src/Components/Main.js

```diff
@@ -3,34 +3,50 @@ import { perform } from '../API'
 import './Main.css';
 import Login from './Login.js';
 
-
 class Main extends Component {
 
   state = {
     player: null,
     message: ''
-  }
+  };
 
   async move(direction) {
-    let results = await perform('move', {direction}).then(res => res.json())
-    if(results.ok) {
-      this.setState({player: results.player, message: results.message})
+    let results = await perform('move', { direction }).then(res => res.json())
+    if (results.ok) {
+      this.setState({ player: results.player, message: results.message })
+    } else if (results.message != null) {
+      this.setState({ message: results.message })
+    }
+  }
+
+  async punch() {
```

Comment:
Is this scalable? There could be anywhere from 2->5 actions someone could perform in combat

Reply:
I mean if it's only 2-5 that's pretty fine imo. In general it would be good to make this data driven if we have lots of actions. For now I think it's fine just to do it and re-factor as we add more.